src/linear_algebra/free_module.lean

```diff
@@ -129,6 +129,12 @@ instance pi {ι : Type*} [fintype ι] {M : ι → Type*} [Π (i : ι), add_comm_
 [Π (i : ι), module R (M i)] [Π (i : ι), module.free R (M i)] : module.free R (Π i, M i) :=
 of_basis $ pi.basis $ λ i, choose_basis R (M i)
 
+/-- Any module is a quotient of a free module. This is stated as surjectivity of
+`finsupp.total M M R id : (M →₀ R) →ₗ[R] M`. -/
+lemma resolution (M : Type*) [add_comm_monoid M] [module R M] :
```

Comment:
I would search for the name `finsupp.total_surjective`, if I were looking for this result.

Reply:
I agree that this is a more natural name thinking about `finsupp`, but I am not sure anybody using `finsupp` will even need it. Indeed, a slightly more general version has been removed by @semorrison in #7734 (saying "you'd never have a surjective function from an indexing set to a module"). Maybe it is better to state it like "there exist a free module such that..."?

Reply:
Should we add back that lemma, but with a docstring that mentions this use case?

Reply:
Adding the lemma back is reasonable, but I am not sure someone playing with free modules will find it easily (unless we keep it in `free_module.lean`.

Reply:
I think it's ok to add it back. And I would even add it back in the original location. The docstring can explain why it exists. And In this file we can add a pointer to the lemma.

Reply:
I've added `total_id_surjective` in `linear_algebra/finsupp.lean`, and upgraded the docstring in `linear_algebra/free_module.lean`